packages/app-review/ios/Plugin/AppReview.swift

```diff
@@ -1,8 +1,35 @@
 import Foundation
+import Capacitor
+import StoreKit
 
 @objc public class AppReview: NSObject {
-    @objc public func echo(_ value: String) -> String {
-        print(value)
-        return value
+    private let plugin: AppReviewPlugin
+    private let requestCountKey = "reviewRequestCount"
+
+    init(plugin: AppReviewPlugin) {
+        self.plugin = plugin
+    }
+
+    @objc public func openAppStore(appID: String) {
+        if let url = URL(string: "https://apps.apple.com/app/id\(appID)?action=write-review") {
+            UIApplication.shared.open(url, options: [:], completionHandler: nil)
+        } else {
+            CAPLog.print("Could not open App Store with appID: \(appID)")
+        }
+
+    }
+
+    @objc public func requestReview() {
+        DispatchQueue.main.async {
```

Comment:
We should not resolve the call if the operation is still in progress. The `requestReview()` method returns before the code in `DispatchQueue.main.async` has finished executing. This is because `DispatchQueue.main.async` uses a callback. Please pass a `completion` function `requestReview()` which can be called in `DispatchQueue.main.async`. This way you can also trigger an error if this method is not available on this device.

Example: https://github.com/capawesome-team/capacitor-plugins/blob/d4e76b7f64b165b703667bfb0e5e67e2a543f7e8/packages/live-update/ios/Plugin/LiveUpdatePlugin.swift#L31

https://github.com/capawesome-team/capacitor-plugins/blob/d4e76b7f64b165b703667bfb0e5e67e2a543f7e8/packages/live-update/ios/Plugin/LiveUpdate.swift#L46

Also, please create properties for all possible errors, example: https://github.com/capawesome-team/capacitor-plugins/blob/d4e76b7f64b165b703667bfb0e5e67e2a543f7e8/packages/datetime-picker/ios/Plugin/DatetimePickerPlugin.swift#L10